frontend/src/pages/NewRun.test.tsx

```diff
@@ -0,0 +1,1160 @@
+        ],
+      });
+      tree.unmount();
+    });
+
+    it('includes additional trigger input fields if run will be recurring', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.isRecurring}=1`;
+      const tree = shallow(<TestNewRun {...props as any} />);
+      await TestUtils.flushPromises();
+
+      expect(tree).toMatchSnapshot();
+      tree.unmount();
+    });
+
+    it('sends a request to create a new recurring run when \'Create\' is clicked', async () => {
+
+      const props = generateProps();
+      const experiment = newMockExperiment();
+      const pipeline = newMockPipeline();
+      props.location.search =
+        `?${QUERY_PARAMS.isRecurring}=1`
+        + `&${QUERY_PARAMS.experimentId}=${experiment.id}`
+        + `&${QUERY_PARAMS.pipelineId}=${pipeline.id}`;
+
+      getExperimentSpy.mockImplementation(() => experiment);
+      getPipelineSpy.mockImplementation(() => pipeline);
+
+      const tree = TestUtils.mountWithRouter(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'test run name' } });
+      (tree.instance() as any).handleChange('description')({ target: { value: 'test run description' } });
+      await TestUtils.flushPromises();
+
+      tree.find('#createNewRunBtn').at(0).simulate('click');
+      // The create APIs are called in a callback triggered by clicking 'Create', so we wait again
+      await TestUtils.flushPromises();
+
+      expect(createRunSpy).toHaveBeenCalledTimes(0);
+      expect(createJobSpy).toHaveBeenCalledTimes(1);
+      expect(createJobSpy).toHaveBeenLastCalledWith({
+        description: 'test run description',
+        enabled: true,
+        max_concurrency: '10',
+        name: 'test run name',
+        pipeline_spec: {
+          parameters: pipeline.parameters,
+          pipeline_id: pipeline.id,
+        },
+        resource_references: [{
+          key: {
+            id: experiment.id,
+            type: ApiResourceType.EXPERIMENT,
+          },
+          relationship: ApiRelationship.OWNER,
+        }],
+        // Default trigger
+        trigger: {
+          periodic_schedule: {
+            end_time: undefined,
+            interval_second: '60',
+            start_time: undefined,
+          },
+        },
+      });
+      tree.unmount();
+    });
+
+    it('displays an error message if periodic schedule end date/time is earlier than start date/time', async () => {
+      const props = generateProps();
+      props.location.search =
+      `?${QUERY_PARAMS.isRecurring}=1`
+      + `&${QUERY_PARAMS.experimentId}=${newMockExperiment().id}`
+      + `&${QUERY_PARAMS.pipelineId}=${newMockPipeline().id}`;
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'test run name' } });
+      // TODO: figure out how to do this by interacting with the Trigger element
```

Comment:
No need, I think this is what the test should do. Trigger component should be tested separately using its interactions.